src/lib/sedlexing.ml

```diff
@@ -117,48 +140,84 @@ let refill lexbuf =
   end
   else lexbuf.len <- lexbuf.len + n
 
+let new_line lexbuf =
+  if lexbuf.curr_line != 0 then
+  lexbuf.curr_line <- lexbuf.curr_line + 1;
+  lexbuf.curr_bol <- lexbuf.pos + lexbuf.offset
+
```

Comment:
This function should probably be exposed in the API.

Reply:
Should it? If so, someone from ocmallex might be confused if they expect `next` to **not** handle new_line automatically and use this incorrectly.
Maybe we should expose it later with some other api that provide ability to config line_ending. Set line_ending to empty list indicates handle the line ending by user. (And it worth another PR)

Reply:
Fair enough. In all cases, the fact that line handling is automatic should be documented!